tests/Matcher/JsonMatcherTest.php

```diff
@@ -113,6 +113,26 @@ public function test_error_when_path_in_nested_value_does_not_exist()
         $this->assertEquals($this->matcher->getError(), 'There is no element under path [foo][bar][faz] in value.');
     }
 
+    public function test_error_when_json_pattern_is_invalid()
+    {
+        $value = '{"test": "value"}';
+        $pattern = '{"test": "@string@",}';
+
+        $this->assertFalse($this->matcher->match($value, $pattern));
+
+        $this->assertEquals($this->matcher->getError(), 'Invalid given JSON of pattern. Syntax error, malformed JSON');
+    }
+
+    public function test_error_when_json_value_is_invalid()
+    {
+        $value = '{"test": "value",}';
+        $pattern = '{"test": "@string@"}';
+
+        $this->assertFalse($this->matcher->match($value, $pattern));
+
+        $this->assertEquals($this->matcher->getError(), 'Invalid given JSON of value. Syntax error, malformed JSON');
```

Comment:
is there any way to add also tests for `JSON_ERROR_DEPTH`, `JSON_ERROR_STATE_MISMATCH`, `JSON_ERROR_CTRL_CHAR` & `JSON_ERROR_UTF8` json errors??